samples: immediate logging  mode in samples nvs

FRASTM opened the pull request:
Use the immediate logging mode in that **samples/subsys/nvs**
This PR is similar to https://github.com/zephyrproject-rtos/zephyr/pull/52949/

It makes 
- samples/drivers/watchdog : fixed by https://github.com/zephyrproject-rtos/zephyr/pull/52949/
- samples/subsys/task_wdt : fixed by https://github.com/zephyrproject-rtos/zephyr/pull/52949/
- samples/subsys/nvs
PASSED 

Fixes #52709 

Signed-off-by: Francois Ramu <francois.ramu@st.com>

martinjaeger commented:
For my understanding: Why is this required?

In the linked PR the argument was that in a test that tests the logging subsystem itself it should not be used by the application for other purposes (printk messages) at the same time. I don't see this being the case for watchdog and NVS.

erwango reviewed (dismissed):
Samples' prj.conf should only contain configuration related to how to use a specific subsystem.
If configuration related to twister is required there is a bug somewhere else IMO

FRASTM commented:
Changing printk with **printf** in src/main.c sample application instead  makes the testcases PASSED as well

FRASTM committed:
samples: Use immediate logging mode in nvs sample

Applying same fix as for watchdog relmated samples
in https://github.com/zephyrproject-rtos/zephyr/pull/52949/
Switching to the immediate logging mode, not by just disabling the
LOG_PRINTK option.Do not direct printk to logging subsystem in the sample.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
samples PASSED with twister without CONFIG_LOG=y

- and  w/o CONFIG_WDT_LOG_LEVEL_DBG=y in the samples/subsys/task_wdt/prj.conf
- and  w/o CONFIG_NVS_LOG_LEVEL_DBG=y in the  samples/subsys/nvs/prj.conf  
- and  w/o CONFIG_WDT_LOG_LEVEL_DBG=y in the samples/drivers/watchdog/prj.conf
-

FRASTM commented:
Still failing : zephyr/samples/subsys/nvs/sample.nvs.basic :
```
Rebooting in ...561 62 --- 115 messages dropped ---
63 64 65 66 67 68 69 6a 6b 6c 6d 6e 6f 70 71 72 73 74 75 76 77 78 79 7a 7b 7c 7d 7e 7f 
...28--- 79 messages dropped ---
...27...26...25...24...23...22...21...20...19...18...17...16...15...14...13...12...11...10...9...8...7...6...5...4...3...2...1...0
Oldest reboot counter: 0
Rebooting in ...561 62 --- 115 messages dropped ---
63 64 65 66 67 68 69 6a 6b 6c 6d 6e 6f 70 71 72 73 74 75 76 77 78 79 7a 7b 7c 7d 7e 7f 
...28--- 80 messages dropped ---
...27...26...25...24...23...22...21...20...19...18...17...16...15...14...13...12...11...10...9...8...7...6...5...4...3...2...1...0
Oldest reboot counter: 0
Rebooting in ...561 62 --- 115 messages dropped ---
63 64 65 66 67 68 69 6a 6b 6c 6d 6e 6f 70 71 72 73 74 75 76 77 78 79 7a 7b 7c 7d 7e 7f 
...28--- 81 messages dropped ---
...27...26...25...24...23...22...21...20...19...18...17...16...15...14...13...12...11...10...9...8...7...6...5...4...3...2...1...0
Oldest reboot counter: 0
Rebooting in ...561 62 --- 115 messages dropped ---
63 64 65 66 67 68 69 6a 6b 6c 6d 6e 6f 70 71 72 73 74 75 76 77 78 79 7a 7b 7c 7d 7e 7f 
...28--- 82 messages dropped ---
...27...26...25...24...23...22...21...20...19...18...17...16...15...14...13...12...11...10...9...8...7...6...5...4...3...2...1...0
```

FRASTM commented:
Change the prj.conf  with  LOG_MODE_IMMEDIATE to have the zephyr/samples/subsys/nvs/sample.nvs.basic PASSED: 

*** Booting Zephyr OS build zephyr-v3.2.0-2493-gfe7ab0939d54 ***
[00:00:00.005,000] <dbg> fs_nvs: nvs_recover_last_ate: Recovering last ate from sector 0
[00:00:00.016,000] <dbg> fs_nvs: nvs_add_gc_done_ate: Adding gc done ate at 3f0
[00:00:00.024,000] <inf> fs_nvs: 3 Sectors of 1024 bytes
[00:00:00.030,000] <inf> fs_nvs: alloc wra: 0, 3e8
[00:00:00.035,000] <inf> fs_nvs: data wra: 0, 0
No address found, adding 192.168.1.1 at id 1
No key found, adding it at id 2
No Reboot counter found, adding it at id 3
Id: 4 not found, adding it
Longarray not found, adding it as id 5
Reboot counter history: ...0
Oldest reboot counter: 0
Rebooting in ...5...4...3...2...1
```
